datahub-web-react/src/app/entity/shared/utils.ts

```diff
@@ -83,14 +85,43 @@ export const isListSubset = (l1, l2): boolean => {
     return l1.every((result) => l2.indexOf(result) >= 0);
 };
 
+function normalize(value: string) {
+    return value.trim().toLowerCase();
+}
+
+function fromQueryGetBestMatch(selectedMatchedFields: MatchedField[], rawQuery: string) {
+    const query = normalize(rawQuery);
+    // first lets see if there's an exact match between a field value and the query
+    const exactMatch = selectedMatchedFields.find((field) => normalize(field.value) === query);
+    if (exactMatch) {
+        return exactMatch;
+    }
+
+    // if no exact match exists, we'll see if the entire query is contained in any of the values
+    const containedMatch = selectedMatchedFields.find((field) => normalize(field.value).includes(query));
+    if (containedMatch) {
+        return containedMatch;
+    }
+
+    // otherwise, just return whichever is first
+    return selectedMatchedFields[0];
+}
+
 export const getMatchPrioritizingPrimary = (
     matchedFields: MatchedField[],
     primaryField: string,
 ): MatchedField | undefined => {
-    const primaryMatch = matchedFields.find((field) => field.name === primaryField);
-    if (primaryMatch) {
-        return primaryMatch;
+    const { location } = window;
+    const params = QueryString.parse(location.search, { arrayFormat: 'comma' });
+    const query: string = decodeURIComponent(params.query ? (params.query as string) : '');
+
+    const primaryMatches = matchedFields.filter((field) => field.name === primaryField);
+    if (primaryMatches.length > 0) {
+        return fromQueryGetBestMatch(primaryMatches, query);
     }
 
-    return matchedFields.find((field) => FIELDS_TO_HIGHLIGHT.has(field.name));
+    return fromQueryGetBestMatch(
+        matchedFields.filter((field) => FIELDS_TO_HIGHLIGHT.has(field.name)),
```

Comment:
nit: it would be cleaner to give the first arg to this function a descriptive name